### PR TITLE
Basic idam logic added

### PR DESCRIPTION
**Pull request by divisathyan**
Basic IDAM classes added

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/Event.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+
+public enum Event {
+
+    START_APPEAL("startAppeal"),
+    EDIT_APPEAL("editAppeal"),
+    SUBMIT_APPEAL("submitAppeal"),
+    SEND_DIRECTION("sendDirection"),
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/Event.java:10]: We don't need all these Events.  Maybe just keep 1 or 2 like `submitAppeal` and remove the rest

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    @Override
+    public void configure(WebSecurity web) {
+        web.ignoring().mvcMatchers(
+            anonymousPaths
+                .stream()
+                .toArray(String[]::new)
+        );
+
+        /*web.ignoring().antMatchers("/swagger-ui.html",
+            "/webjars/springfox-swagger-ui/**",
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/SecurityConfiguration.java:50]: remove the comments

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/UserDetailsProviderConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        return new IdamUserDetailsProvider(
+            requestUserAccessTokenProvider,
+            restTemplate,
+            baseUrl,
+            detailsUri
+        );
+    }
+
+    @Bean("systemUser")
+    public UserDetailsProvider getSystemUserDetailsProvider(
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/UserDetailsProviderConfiguration.java:34]: probably don't need this system user here.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/security/idam/IdamAuthorizor.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+public class IdamAuthorizor {
+
+    private final RestTemplate restTemplate;
+    private final String baseUrl;
+    private final String clientId;
+    private final String clientSecret;
+    private final String clientRedirectUri;
+
+    public IdamAuthorizor(
+        RestTemplate restTemplate,
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/security/idam/IdamAuthorizor.java:28]: This is ok for now, but we will have to change this to conform to OpenId

File: src/main/resources/application.yaml [hunk 5]
```diff
@@ -15,6 +15,102 @@ spring:
...
+    - "/health"
+    - "/health/liveness"
+    - "/loggers/**"
+    - "/swagger-ui.html"
+    - "/swagger-resources/**"
+    - "/v2/api-docs"
+    - "/webjars/springfox-swagger-ui/**"
+    - "/favicon.ico"
+  authorisedServices:
+    - "ccd"
```
Comment (raghera) [inline src/main/resources/application.yaml:33]: only `iac` is an authorized service

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -15,6 +15,102 @@ spring:
...
+    - "caseworker-ia"
+    - "citizen"
+  roleEventAccess:
+    citizen:
+      - "startAppeal"
+      - "editAppeal"
+      - "submitAppeal"
+      - "submitReasonsForAppeal"
+    caseworker-ia-legalrep-solicitor:
+      - "startAppeal"
```
Comment (raghera) [inline src/main/resources/application.yaml:48]: again we don't need all these events.  We only care about events that will trigger the home-office api which is likely to be submitAppeal.  Keep it lean for now, we can add any others we need later easily

File: src/main/resources/bootstrap.yaml [hunk 7]
```diff
@@ -4,6 +4,8 @@ spring:
       prefixed: false
       paths: /mnt/secrets/ia
       aliases:
+        system-username: IA_SYSTEM_USERNAME
```
Comment (raghera) [inline src/main/resources/bootstrap.yaml:7]: don't need a system user here.

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/EventTest.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class EventTest {
+
+    @Test
+    public void has_correct_values() {
+        assertEquals("startAppeal", Event.START_APPEAL.toString());
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/EventTest.java:12]: This test will need changing if you remove some events.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/Event.java (hunk 1 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/Event.java]: Removed all other events except submitAppeal

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/SecurityConfiguration.java (hunk 2 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/SecurityConfiguration.java]: Removed comment

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/UserDetailsProviderConfiguration.java (hunk 3 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/UserDetailsProviderConfiguration.java]: Removed systemUser

File: src/main/resources/application.yaml (hunk 5 above)
Comment (divisathyan) [reply src/main/resources/application.yaml]: Removed other values

File: src/main/resources/application.yaml (hunk 6 above)
Comment (divisathyan) [reply src/main/resources/application.yaml]: Removed all other events except submitAppeal

File: src/main/resources/bootstrap.yaml (hunk 7 above)
Comment (divisathyan) [reply src/main/resources/bootstrap.yaml]: Removed systemUser

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/EventTest.java (hunk 8 above)
Comment (divisathyan) [reply src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/entities/ccd/EventTest.java]: Removed all other events except submitAppeal

**Pull request merged**